### DISCUSSION TITLE
Integrasi google form ke whatsapp bermasalah

### DISCUSSION THREAD

**Discussion by endrianur** [kirimwa-id/forum#168] [Tanya Jawab] [answered]
saya mendapatkan hasil log seperti dibawah ini :

![errorgakekirim](https://user-images.githubusercontent.com/126568698/221921732-9cd37754-6961-4acf-8698-9ad2e60b4785.png)

skrip lengkapnya seperti ini : 

```js
function FormSubmit(e) {
    var formResponse = e.response;
    var items = formResponse.getItemResponses();
  
    var answer = {};
  
    for (var i = 0; i < items.length; i++) {
      var question = items[i].getItem().getTitle().toLowerCase().replace(/[\W\-_]/g, ' ').trim().replace(/ /g, '_').toLowerCase();
      var response = items[i].getResponse();
  
      answer[question] = response;
      answer['timestamp'] = formResponse.getTimestamp().toISOString();
  
      Logger.log(answer);
    }
  
    var data = {
      message: `Hello *${answer.nama}*,\n\nTerima kasih telah membeli. Mohon tunggu konfirmasi dari Admin untuk langkah selanjutnya.`,
      phone_number: answer.nomor_whatsapp
    }
    sendWhatsAppResponse(data);
  
    var adminData = {
      message: `*Terdapat Pembelian Baru*\n\nNama: ${answer.nama}\nWhatsApp: ${answer.nomor_whatsapp}\nBuka chat: https://kirimwa.id/` + toInternationalFormat(answer.nomor_whatsapp),
      phone_number: '6281314879248'
    }
    sendWhatsAppResponse(adminData);
  }
  
  /**
   * https://developer.kirimwa.id/api/v1/
   */
  function sendWhatsAppResponse(data)
  {
    var api = {
      url: "https://api.kirimwa.id/v1/messages",
      token: "xxxREwl7cG9z80mkVfFM/rkZiIV8Myvg4BFRqdm1QTDLVxMGWJX3thv2eUXXXX"
    }
  
    var payload = {
      message: data.message,
      phone_number: toInternationalFormat(data.phone_number),
      device_id: "XXB3C19C273XXXXX",
      // optional parameters
      message_type: "text",
      send_at: "now",
      is_group_message: false,
    }
  
    var response = UrlFetchApp.fetch(api.url, {
      method: "post",
      contentType: "application/json",
      headers: {
        authorization: "Bearer " + api.token
      },
      payload: JSON.stringify(payload)
    });
  
    Logger.log(response);
  }
  
  function toInternationalFormat(phone)
  {
    return phone.replace(/[^0-9]/g, '').replace(/^0|^62/, 62);
  }
```

Hasilnya, setelah submit di google form tidak terkirim ke whatsapp.

Ada yang bisa bantu?

**Comment by endrianur** [comment]
@rioastamal

**Reply by rioastamal** [reply]
Tampilan form-nya seperti apa?

**Reply by endrianur** [reply]
![form](https://user-images.githubusercontent.com/126568698/222023745-1532290d-dc08-48bf-98a3-a65599a11871.png)

**Reply by endrianur** [reply]
error masuk ke email 

![errorkirimwa_Gmail](https://user-images.githubusercontent.com/126568698/222055437-62a140cd-d370-4e4c-adde-23b48c9fedb0.jpg)


device_id sdh benar pake aplikasi device id = (android device id), cuma masih error ya, devicenya ga ke detect.

Please help....
@rioastamal

**Comment by endrianur** [comment]
Belum ada solusinya kah?

Apakah bermasalah dengan servernya? karna ada pembaharuan google script?

**Reply by melodiazz** [reply]
sesuai pesannya 404 device not found, kemungkinan ada kesalahan dalam mendefinisikan nama device

**Reply by endrianur** [reply]
device id sudah sesuai dengan menggunakan aplikasi device id. cuma cara memperbaikinya gimana?apakah tidak ada yang sudah bisa menjalankannya?

jika pake whatstarget sudah bisa tapi dibatasi hanya 20 pesan saja

**Reply by kafaa-dev** [reply]
sudah buat device nya belum?

**Reply by endrianur** [reply]
> sudah buat device nya belum?

buat device?bukannya devicenya HP yg akan digunakan?

**Reply by melodiazz** [reply]
https://developer.kirimwa.id/api/v1/#post-devices

**Reply by endrianur** [reply]
sdh bisa berhasil?boleh share kode.gs nya?

**Reply by endrianur** [reply]
> sudah buat device nya belum?

caranya gmn?ada scriptnya yg bs dicopy?

**Reply by endrianur** [reply]
> https://developer.kirimwa.id/api/v1/#post-devices

terima kasih sudah bisa....

pesan pending...

tapi di cek pesannya kok ga ada ya?
message not found

padahal pas dikirim :
{
    "id": "kwid-e0b995bf9b834c928e2baa62bbf",
    "status": "pending",
    "message": "Message is pending and waiting to be processed.",
    "meta": {
        "location": "https://api.kirimwa.id/v1/messages/kwid-e0b995bf9b834c928e2baa62bbf"
    }
}

**Reply by endrianur** [reply]
ups sorry...ternyata kwidnya lupa ditaruh....done....message sent

**Comment by endrianur** [comment, marked as answer]
![Screenshot_20230304_101849_WhatsApp](https://user-images.githubusercontent.com/126568698/222873743-2ae43e78-713d-4ba0-9af6-3acac441d0e6.jpg)


done...sudah berhasil...

solved.
